markdownlint.js

```diff
@@ -162,12 +163,17 @@ function printResult(lintResult) {
   }
 
   if (program.output) {
-    try {
-      fs.writeFileSync(program.output, lintResultString);
-    } catch (error) {
-      console.warn('Cannot write to output file ' + program.output + ': ' + error.message);
-      process.exitCode = 2;
-    }
+      try {
+        if (lintResultString.length > 0) {
+          fs.writeFileSync(program.output, lintResultString + os.EOL); // We need to append a newline to make a valid POSIX file.
```

Comment:
Could this become a 1-line change with something like:
`fs.writeFileSync(program.output, lintResultString + (lintResultString.length > 0 ? os.EOL : '');`

Reply:
The meaning is not as clear, and it's use is [contentious](https://eslint.org/docs/rules/no-ternary). Many maintainers disallow it, so I default to the longer form.

I've refactored the code to use a long-form ternary before we attempt to write out. It should be a bit better. Let me know what you think!